### PR TITLE
ACTIONSCRIPT: Null value handling and other improvements

### PR DESCRIPTION
**Pull request by Nostritius**
This PR adds the specific null value to actionscript and adds some other improvements

### CODE REVIEW

File: src/aurora/actionscript/asbuffer.cpp [hunk 1]
```diff
@@ -464,7 +464,20 @@ void ASBuffer::actionCallMethod(AVM &avm) {
...
 
 void ASBuffer::actionEnumerate2() {
-	// TODO
+	if (!_stack.top().isObject())
+		throw Common::Exception("actionEnumerate2: not an object");
+
+	ObjectPtr object = _stack.top().asObject();
+	_stack.pop();
+
+	// push null as the end of the enumeration.
```
Comment (DrMcCoy) [inline src/aurora/actionscript/asbuffer.cpp]: Please start a comment with an upper-case letter. And no period at the end of simple single-line comments

File: src/aurora/actionscript/object.cpp [hunk 2]
```diff
@@ -44,6 +44,14 @@ Object::Object(Object *object) {
 Object::~Object() {
 }
 
+std::vector<Common::UString> Object::getSlots() {
```
Comment (DrMcCoy) [inline src/aurora/actionscript/object.cpp]: That's a const method

File: src/aurora/actionscript/asbuffer.cpp [hunk 3]
```diff
@@ -464,7 +464,20 @@ void ASBuffer::actionCallMethod(AVM &avm) {
 }
 
 void ASBuffer::actionEnumerate2() {
-	// TODO
+	if (!_stack.top().isObject())
+		throw Common::Exception("actionEnumerate2: not an object");
+
+	ObjectPtr object = _stack.top().asObject();
```
Comment (DrMcCoy) [inline src/aurora/actionscript/asbuffer.cpp]: That should be const

File: src/aurora/actionscript/asbuffer.cpp [hunk 4]
```diff
@@ -268,8 +268,16 @@ void ASBuffer::actionTrace() {
 	debugC(kDebugActionScript, 1, "actionTrace");
 }
 
-void ASBuffer::actionDefineLocal() {
-	// TODO
+void ASBuffer::actionDefineLocal(AVM &avm) {
+	Variable value = _stack.top();
+	_stack.pop();
+	Variable name = _stack.top();
```
Comment (DrMcCoy) [inline src/aurora/actionscript/asbuffer.cpp]: These two Variables are const

File: src/aurora/actionscript/string.cpp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  String implementation for actionscript.
+ */
+
+#include "string.h"
```
Comment (DrMcCoy) [inline src/aurora/actionscript/string.cpp]: Full paths, please.

File: src/aurora/actionscript/string.cpp [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace ActionScript {
+
+String::String() {
+}
+
+} // End of namespace ActionScript
+
+} // End of namespace Aurora
+
```
Comment (DrMcCoy) [inline src/aurora/actionscript/string.cpp]: That's one newline too many

File: src/aurora/actionscript/textfield.cpp [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace ActionScript {
+
+TextField::TextField() {
+}
+
+} // End of namespace ActionScript
+
+} // End of namespace Aurora
+
```
Comment (DrMcCoy) [inline src/aurora/actionscript/textfield.cpp]: One newline too many

File: src/aurora/actionscript/asbuffer.cpp [hunk 8]
```diff
@@ -504,6 +506,25 @@ void ASBuffer::actionExtends() {
...
+		throw Common::Exception("actionGetURL: url is not a string");
+	_stack.pop();
+	Variable target = _stack.top();
+	if (target.isString())
+		throw Common::Exception("actionGetURL: target is not a string");
+	_stack.pop();
+
+	Common::UString urlString, targetString;
+	urlString = url.asString();
+	targetString = target.asString();
```
Comment (DrMcCoy) [inline src/aurora/actionscript/asbuffer.cpp]: The Variables and the strings are all const :P

**Comment by Nostritius** [comment]
The issues should now be fixed

**Comment by DrMcCoy** [comment]
Merged as abcde6d63...bf4c5a39e, thanks! :)

**Pull request closed without merging**